internal/pkg/cli/env_upgrade.go

```diff
@@ -88,9 +109,70 @@ func (o *envUpgradeOpts) Ask() error {
 // Execute updates the cloudformation stack an environment to the specified version.
 // If the environment stack is busy updating, it spins and waits until the stack can be updated.
 func (o *envUpgradeOpts) Execute() error {
+	envs, err := o.listEnvsToUpgrade()
+	if err != nil {
+		return err
+	}
+	for _, env := range envs {
+		if err := o.upgrade(env); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *envUpgradeOpts) listEnvsToUpgrade() ([]string, error) {
+	if !o.all {
+		return []string{o.name}, nil
+	}
+
+	envs, err := o.store.ListEnvironments(o.appName)
+	if err != nil {
+		return nil, fmt.Errorf("list environments in app %s: %v", o.appName, err)
+	}
+	var names []string
+	for _, env := range envs {
+		names = append(names, env.Name)
+	}
+	return names, nil
+}
+
+func (o *envUpgradeOpts) upgrade(env string) error {
+	yes, err := o.shouldUpgrade(env)
+	if err != nil {
+		return err
+	}
+	if !yes {
+		return nil
+	}
 	return nil
 }
 
+func (o *envUpgradeOpts) shouldUpgrade(env string) (bool, error) {
+	envTpl, err := o.newEnvVersionGetter(o.appName, env)
+	if err != nil {
+		return false, err
+	}
+	version, err := envTpl.Version()
+	if err != nil {
+		return false, fmt.Errorf("get template version of environment %s in app %s: %v", env, o.appName, err)
+	}
+
+	diff := semver.Compare(version, deploy.LatestEnvTemplateVersion)
+	if diff < 0 {
+		// Newer version available.
+		return true, nil
+	}
+
+	msg := fmt.Sprintf("Environment %s is already on the latest version %s, skip upgrade.", env, deploy.LatestEnvTemplateVersion)
+	if diff > 0 {
+		msg = fmt.Sprintf(`Skip upgrading environment %s to version %s since it's on version %s. 
+Are you using the latest version of AWS Copilot?`, env, deploy.LatestEnvTemplateVersion, version)
+	}
+	log.Debugln(msg)
+	return false, nil
+}
+
 // buildEnvUpgradeCmd builds the command to update the
```

Comment:
Don't leave me hanging! The suspense is killing me!

Reply:
lool sorry ahahha i'll fix these in the next PR :D